test/jdk/java/awt/Multiscreen/MultiScreenCheckScreenIDTest.java

```diff
@@ -0,0 +1,144 @@
+import java.util.ArrayList;
+import java.util.List;
+
+/*
+ * @test
+ * @bug 8280482
+ * @key headful
+ * @requires (os.family == "linux")
+ * @library /java/awt/regtesthelpers
+ * @summary Test to check if window GC doesn't change within same screen.
+ * @run main MultiScreenCheckScreenIDTest
+ */
+
+public class MultiScreenCheckScreenIDTest {
+    private static final int COLS = 12;
+    private static final Color BACKGROUND = new Color(0, 0, 255, 64);
+    private static GraphicsDevice[] screens;
+    static List<Window> windowList = new ArrayList<>();
+
+    static Robot robot;
+
+    public static void main(final String[] args) throws Exception {
+        try {
+            createGUI();
+            robot.delay(100);
+            robot.waitForIdle();
+        } finally {
+            for (Window win : windowList) {
+                win.dispose();
+            }
+        }
+        System.out.println("Test Pass");
+    }
+
+    private static void createGUI() throws AWTException {
+        new MultiScreenCheckScreenIDTest().createWindowGrid();
+    }
+
+    private void createWindowGrid() throws AWTException {
+        screens = GraphicsEnvironment
+                .getLocalGraphicsEnvironment()
+                .getScreenDevices();
+
+        if (screens.length < 2) {
+            System.out.println("Testing aborted. Required min of 2 screens. " +
+                    "Found : " + screens.length);
+            return;
+        }
+        robot = new Robot();
+
+        for (GraphicsDevice screen : screens) {
+            Rectangle screenBounds = screen.getDefaultConfiguration().getBounds();
+
+            for (Rectangle r : gridOfRectangles(screenBounds, COLS)) {
+                try {
+                    SwingUtilities.invokeAndWait(() -> {
+                        try {
+                            createWindow(r);
+                        } catch (Exception e) {
+                            throw new RuntimeException(e);
+                        }
+                    });
+                } catch (InterruptedException | InvocationTargetException e) {
+                    e.printStackTrace();
+                }
+                robot.delay(50);
+                robot.waitForIdle();
+                if (windowList.get(windowList.size() - 1).getBounds().intersects
+                        (screens[0].getDefaultConfiguration().getBounds())) {
+                    if (!(windowList.get(windowList.size() - 1).
+                            getGraphicsConfiguration().getBounds().
+                            intersects(screens[0].getDefaultConfiguration().
+                                    getBounds()))) {
+                        throw new RuntimeException("Graphics configuration " +
+                                "changed for screen 0");
+                    }
+                } else if (windowList.get(windowList.size() - 1).getBounds().
+                        intersects(screens[1].getDefaultConfiguration().getBounds())) {
```

Comment:
What if user has more than 2 screens?
It looks like the 3rd and subsequent screens are not tested.

Reply:
I have made the validation as per number of screens now. I guess this is fine to check multiple screens.

Reply:
I don't see why we should be limited to only two when more may be available.
What if it helps to find some issue?
Anyway, the test place windows on all screens, but checking only on two of them.

Reply:
I have updated the test to available screens (Though not able to test since I can extend to two screens now), I guess it should work, it's generic now.

Reply:
FYI, you can arrange up to 8 screens using Virtual Box(but I don't think it's necessary in this case)